src/Compiler/Microsoft.AspNetCore.Razor.Language/src/MetadataCollection.cs

```diff
@@ -409,24 +387,67 @@ public override bool Equals(MetadataCollection other)
 
         protected override int ComputeHashCode()
         {
-            var hash = HashCodeCombiner.Start();
-
-            hash.Add(_key1, StringComparer.Ordinal);
-            hash.Add(_value1, StringComparer.Ordinal);
+            return _count switch
+            {
+                1 => ComputeHashCodeForOneItem(),
+                2 => ComputeHashCodeForTwoItems(),
+                _ => ComputeHashCodeForThreeItems()
```

Comment:
I didn't want to throw because we've already guaranteed that count can only be "1", "2", or "3" and throwing an exception can cause the JIT not to inline.